app.js

```diff
@@ -487,31 +489,21 @@ d.run(() => {
 					});
 			},
 
-			/**
-			 * Description of the function.
-			 *
-			 * @memberof! app
-			 * @param {function} cb
-			 * @todo Add description for the params
-			 */
-			cache(cb) {
-				const RedisConnector = require('./helpers/redis_connector.js');
-				logger.debug(
-					`Cache ${appConfig.cacheEnabled ? 'Enabled' : 'Disabled'}`
-				);
-				// delete password key if it's value is null
-				const cacheConfigParam = Object.assign({}, config.cache);
-				if (cacheConfigParam.password === null) {
-					delete cacheConfigParam.password;
+			components(cb) {
+				logger.debug(`Cache ${config.cacheEnabled ? 'Enabled' : 'Disabled'}`);
+				if (!config.cacheEnabled) {
+					return cb();
 				}
-				const redisConnector = new RedisConnector(
-					config.cacheEnabled,
-					cacheConfigParam,
-					logger
-				);
-				redisConnector.connect((redisConnectError, redisClient) =>
-					cb(null, { cacheEnabled: config.cacheEnabled, client: redisClient })
-				);
+				const cache = createCacheComponent(config.cache, logger);
```

Comment:
Why there is there a need for a separate factory`createCacheComponent` and the `bootstrap` method? They will be called always together, meaning bootstrap can be merged into the factory.

Reply:
`factorycreateCacheComponent` is a synchronous call to create the object. `bootstrap` is to establish the physical connection with redis server. Its intended to have it this way, to bootstrap later in the stack and create object in start of the file.